code/handlers/trackservers.q

```diff
@@ -332,15 +332,22 @@ startup:{
 // Check if required processes all connected
 reqprocsnotconn:{[requiredprocs] 
     not all requiredprocs in exec proctype from .servers.SERVERS where .dotz.liveh[w]
-  }
+  };
 
 // Block process until all required processes are connected
-startupdependent:{[requiredprocs;timeintv]
-    while[.servers.reqprocsnotconn[requiredprocs];
-          .os.sleep[timeintv];
-          .servers.startup[]
-         ]
-  }
+startupdepcycles:{[requiredprocs;timeintv;cycles] 
+  n:0;                                                                                                                  //variable used to check how many cycles have passed
+  while[.servers.reqprocsnotconn requiredprocs;                                                                         //check if requiredprocs are running
+    n+:1;                                                                                                               //cycle counter
+    .servers.startup[];
+    if[n>cycles;
+      b:((),requiredprocs)except(),exec proctype from .servers.SERVERS where .dotz.liveh w;
+      .lg.o[`connectionreport;raze string[.proc.procname]," cannot connect to ",
+      ","sv string@'b];];                                                                                                                //after "cycles" times output error and exit process. 
```

Comment:
`@'` is a bit nasty, can you make this either an `each` or `'[b]` please?

Also not sure this logging needs broken onto 2 lines, might be clearer on one? (I think probably keep the closing bracket for the `if` on this line)